test/he/test_conn.c

```diff
@@ -457,10 +457,14 @@ void test_send_keepalive_connected(void) {
   TEST_ASSERT_EQUAL(HE_SUCCESS, res);
 }
 
-void test_disconnect_reject_if_not_online(void) {
-  wolfSSL_shutdown_IgnoreAndReturn(SSL_FATAL_ERROR);
-  he_return_code_t res = he_conn_disconnect(&conn);
-  TEST_ASSERT_EQUAL(HE_ERR_INVALID_CONN_STATE, res);
+void test_disconnect_reject_if_in_invalid_states(void) {
+  he_conn_state_t states[] = { HE_STATE_NONE, HE_STATE_DISCONNECTED,
+                               HE_STATE_DISCONNECTING, HE_STATE_CONNECTING };
+  for (int i = 0; i < sizeof(states) / sizeof(states[0]); ++i) {
+    conn.state = states[i];
+    he_return_code_t res = he_conn_disconnect(&conn);
+    TEST_ASSERT_EQUAL(HE_ERR_INVALID_CONN_STATE, res);
+  }
```

Comment:
Do we have a test that checks `he_conn_disconnect` actually calls "send goodbye" in valid states?

Reply:
I've refactored the code to make it more easily testable and added a test for this